libsolidity/parsing/Parser.cpp

```diff
@@ -204,7 +204,7 @@ ASTPointer<PragmaDirective> Parser::parsePragmaDirective()
 	nodeFactory.markEndPosition();
 	expectToken(Token::Semicolon);
 
-	if (literals.size() >= 2 && literals[0] == "solidity")
+	if (literals.size() >= 1 && literals[0] == "solidity")
```

Comment:
I was wondering if we should just issue an error here if `literals.size() < 2` instead of calling ``parsePragmaVersion`. We could also make a better error message that the version is missing in the pragma.

Reply:
We could have one error message for `no solidity version given` for the newly added test case. Does seem like an overkill since this is the only test input that will make use of that error category :-)

Reply:
We can add that extra case if people think it is useful.

Reply:
We could further improve this by making `SemVerParser` to return a `util::Result<>` and give specific error messages. I think however we should do that in a separate PR as that is not a bugfix, but extending error reporting capabilities, and it may become rather big of a change.